packages/account/src/providers/utils/extract-tx-error.ts

```diff
@@ -88,12 +95,21 @@ export const assembleRevertError = (
         break;
 
       default:
-        reason = 'unknown';
-        errorMessage = `The transaction reverted with an unknown reason: ${revertReceipt.val}`;
+        throw new FuelError(
+          ErrorCode.UNKNOWN,
```

Comment:
Wouldn't this be best served with a `TRANSACTION_REVERTED`?

Reply:
We currently don't have a `TRANSACTION_REVERTED` error code but it may be worthwhile discussing if there are other similar instances.

I opted for `UNKNOWN` because we are currently unable to determine why the script reverted in this instance. There was a previous discussion in the [above thread](https://github.com/FuelLabs/fuels-ts/pull/2562#discussion_r1647985538)

Reply:
I added my thoughts to the linked conversation.